main.py

```diff
@@ -110,6 +110,34 @@ def main():
 		mouseBall.setPos(moveData["currentPos"])
 		mouseBall.setVelocity(mouseVelocity)
 
+
+		#gravity adjustment
+		if heldKeys[pygame.K_i]:
+			GRAVITY = max(0.0001,GRAVITY+0.1)
+		elif heldKeys[pygame.K_k]:
+			GRAVITY  = max(0.0001,GRAVITY-0.1)
+
+		#ball radius
+		if heldKeys[pygame.K_o]:
+			ball.radius = max(0.1,ball.radius+0.01)
+		elif heldKeys[pygame.K_l]:
+			ball.radius = max(0.1,ball.radius-0.01)
+
+		#string length
+		if heldKeys[pygame.K_u]:
+			string.length = max(0,string.length+0.01)
```

Comment:
Unlike gravity, string length must NOT be 0, otherwise many zero division errors will arise. Maybe 0.01?

Plus the same thing as above, could this be a maximum value?